src/arrow_reader.rs

```diff
@@ -70,6 +73,11 @@ impl<R> ArrowReaderBuilder<R> {
         self
     }
 
+    pub fn with_range(mut self, range: Range<usize>) -> Self {
+        self.range = Some(range);
+        self
+    }
```

Comment:
Needs documentation to explain the exact semantics. Also, I think a more descriptive name than `range` might be needed, such as `file_byte_range`? `range` sounds too generic and could refer to many different things (e.g. could be range of rows)

Reply:
+1